api/proto/auth/client.go

```diff
@@ -0,0 +1,892 @@
+/*
+Copyright 2020 Gravitational, Inc.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+// Package auth implements teleport's grpc auth client
+package auth
+
+import (
+	"compress/gzip"
+	"context"
+	"crypto/tls"
+	"io"
+	"net"
+	"sync"
+	"sync/atomic"
+	"time"
+
+	"github.com/golang/protobuf/ptypes/empty"
+	"github.com/gravitational/teleport"
+	"github.com/gravitational/teleport/lib/events"
+	"github.com/gravitational/teleport/lib/jwt"
+	"github.com/gravitational/teleport/lib/services"
+	"github.com/gravitational/teleport/lib/session"
```

Comment:
I think you can remove the dependence on `services` and `session` in another PR, but what `events`. Can you take a look and tell me if you have remove that as well?

Reply:
I'll do this in a separate PR right now. I think I'll need to move the events proto file along with some interfaces and functions, pretty small refactor.

Reply:
Actually moving the proto file will make this a large change similar to the services refactor, and I can't find a good way to circumvent that, so I think we should table this for a later PR.